processor/src/advice/dsa.rs

```diff
@@ -0,0 +1,118 @@
+use super::{ExecutionError, Felt, StarkField, Vec, Word};
+use vm_core::crypto::dsa::{
+    elements_as_bytes, Polynomial, PublicKeyBytes, SecretKey, SecretKeyBytes, PK_LEN, SK_LEN,
+};
+
+/// Gets as input a vector containing an expanded public key and its associated secret key, and a
+/// word representing a message and outputs a vector of values to be pushed onto the advice stack.
+/// The values are the ones required for a Falcon signature verification inside the VM and they are:
+///
+/// 1. The nonce represented as 8 field elements.
+/// 2. The expanded public key represented as the coefficients of a polynomial of degree < 512.
+/// 3. The signature represented as the coefficients of a polynomial of degree < 512.
+/// 4. The product of the above two polynomials in the ring of polynomials with coefficients
+/// in the Miden field.
+///
+/// Inputs:
+///   Operand stack: [PK, MSG, ...]
+///   Advice stack: [...]
+///   Advice map: {PK: [pk_raw, sk_raw]}
+///
+/// Outputs:
+///   Operand stack: [PK, MSG, ...]
+///   Advice stack: [NONCE1, NONCE0, h, s2, pi]
+///   Advice map: {PK: [pk_raw, sk_raw]}
```

Comment:
I think this can be removed from here, and we may want to update some of the description above to reflect that this just generates a signature (and does not interact with the rest of the VM in any way).